File: csharp/killer-sudoku-helper/KillerSudokuHelper.cs
public static class KillerSudokuHelper
{
    public static IEnumerable<int[]> Combinations(int sum, int size, int[] exclude)
    {
        var result = new List<int[]>();
        var excludeSet = new HashSet<int>(exclude);

        void Backtrack(List<int> currentCombination, int startDigit, int currentSum)
        {
            if (currentCombination.Count == size && currentSum == sum)
            {
                result.Add(currentCombination.ToArray());
                return;
            }
            if (currentCombination.Count >= size || currentSum >= sum)
            {
                return;
            }
            for (int digit = startDigit; digit <= 9; digit++)
            {
                if (exclude.Contains(digit) || currentCombination.Contains(digit))
                {
                    continue;
                }
                currentCombination.Add(digit);
                Backtrack(currentCombination, digit + 1, currentSum + digit);
                currentCombination.RemoveAt(currentCombination.Count - 1);
            }
        }

        Backtrack(new List<int>(), 1, 0);
        return result.OrderBy(comb => string.Join("", comb));
    }
    
}

File: javascript/killer-sudoku-helper/killer-sudoku-helper.js
//
// This is only a SKELETON file for the 'Killer Sudoku Helper' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const combinations = ({ sum, size, exclude}) => {
  const result = [];
  const excludedSet = new Set(exclude);

  function backtrack(start, currentCombo, currentSum) {
    if (currentCombo.length === size) {
      if (currentSum === sum) {
        result.push([...currentCombo]);
      }
      return
    };

    if (currentSum > sum) return;

    for (let i = start; i <= 9; i++) {
      if (excludedSet.has(i)) continue;

      currentCombo.push(i);
      backtrack(i + 1, currentCombo, currentSum + i);
      currentCombo.pop();
    }
  }

  backtrack(1, [], 0);
  return result;
};
